substrate/client/network/sync/src/strategy/state.rs

```diff
@@ -465,6 +481,62 @@ mod test {
 		}
 	}
 
+	#[test]
+	fn backedoff_number_peer_is_not_scheduled() {
+		let client = Arc::new(TestClientBuilder::new().set_no_genesis().build());
+		let target_block = BlockBuilderBuilder::new(&*client)
+			.on_parent_block(client.chain_info().best_hash)
+			.with_parent_block_number(client.chain_info().best_number)
+			.build()
+			.unwrap()
+			.build()
+			.unwrap()
+			.block;
+
+		let peers = (1..=10)
+			.map(|best_number| (PeerId::random(), best_number))
+			.collect::<Vec<(_, _)>>();
+		let ninth_peer = peers[8].0;
+		let tenth_peer = peers[9].0;
+		let initial_peers = peers.iter().map(|(p, n)| (*p, *n));
+
+		let mut state_strategy = StateStrategy::new(
+			client.clone(),
+			target_block.header().clone(),
+			None,
+			None,
+			false,
+			initial_peers,
+		);
+
+		// Disconnecting a peer without an inflight request has no effect on persistent states.
+		state_strategy.remove_peer(&tenth_peer);
+		state_strategy.remove_peer(&tenth_peer);
+		state_strategy.remove_peer(&tenth_peer);
```

Comment:
Is it on purpose to remove the peer three times?

Reply:
Good catch, have removed the peer only once, should have the same testing effect 👍